### PR TITLE
feat(services): migrate resources setting react query

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-920

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Make it work, make it right, make it fast

File: libs/pages/application/src/lib/feature/page-settings-resources-feature/page-settings-resources-feature.tsx [hunk 1]
```diff
@@ -1,116 +1,97 @@
...
-  cloneApplication.memory = Number(data['memory'])
-  cloneApplication.cpu = data['cpu']
-  if (!isJob(application)) {
-    cloneApplication.min_running_instances = data['instances'][0]
-    cloneApplication.max_running_instances = data['instances'][1]
-  }
-
-  return cloneApplication
+export interface SettingsResourcesFeatureProps {
+  service: Exclude<AnyService, Helm | Database>
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/feature/page-settings-resources-feature/page-settings-resources-feature.tsx:12]:
```suggestion
  service: Helm | Database
```

File: libs/pages/application/src/lib/ui/page-settings-resources/page-settings-resources.tsx [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
+import { Button, HelpSection } from '@qovery/shared/ui'
 
 export interface PageSettingsResourcesProps {
   onSubmit: FormEventHandler<HTMLFormElement>
   displayWarningCpu: boolean
-  application?: ApplicationEntity
-  environmentMode?: EnvironmentModeEnum
-  loading?: boolean
-  clusterId?: string
+  service: Exclude<AnyService, Helm | Database>
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/ui/page-settings-resources/page-settings-resources.tsx:10]:
```suggestion
  service: Helm | Database
```

File: libs/shared/console-shared/src/lib/application-settings/ui/application-settings-resources/application-settings-resources.tsx [hunk 3]
```diff
@@ -21,25 +20,30 @@ import {
 
 export interface ApplicationSettingsResourcesProps {
   displayWarningCpu: boolean
-  application?: ApplicationEntity
+  service?: Exclude<AnyService, Helm | Database>
```
Comment (ctjhoa) [inline libs/shared/console-shared/src/lib/application-settings/ui/application-settings-resources/application-settings-resources.tsx:23]:
```suggestion
  service: Helm | Database
```
and you can remove all `service` null checks

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/application/src/lib/feature/page-settings-resources-feature/page-settings-resources-feature.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/pages/application/src/lib/feature/page-settings-resources-feature/page-settings-resources-feature.tsx:12]: ok seens it's Exclude my bad

File: libs/shared/console-shared/src/lib/application-settings/ui/application-settings-resources/application-settings-resources.tsx (hunk 3 above)
Comment (ctjhoa) [reply libs/shared/console-shared/src/lib/application-settings/ui/application-settings-resources/application-settings-resources.tsx:23]: ok seens it's Exclude my bad

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**